### PR TITLE
Color and font inconsistencies fixed

### PR DESCRIPTION
**Pull request by Nirvyz**
Hello,
We fixed several issues related to Navigation Bar color and font colors inconsistencies, detailed in our issue #662 .

### CODE REVIEW

File: android/uhabits-android/src/main/java/org/isoron/uhabits/activities/settings/SettingsActivity.java [hunk 1]
```diff
@@ -52,8 +56,9 @@ private void setupActionBarColor()
         int color = BaseScreen.getDefaultActionBarColor(this);
 
         if (res.getBoolean(R.attr.useHabitColorAsPrimary))
-            color = res.getColor(R.attr.aboutScreenColor);
+            color = res.getColor(R.attr.colorPrimary);
```
Comment (iSoron) [inline android/uhabits-android/src/main/java/org/isoron/uhabits/activities/settings/SettingsActivity.java:59]: Please revert this change. The blue color bar is intended.

File: android/uhabits-android/src/main/java/org/isoron/uhabits/activities/settings/SettingsActivity.java [hunk 2]
```diff
@@ -52,8 +56,9 @@ private void setupActionBarColor()
...
 
         BaseScreen.setupActionBarColor(this, color);
     }
+
```
Comment (iSoron) [inline android/uhabits-android/src/main/java/org/isoron/uhabits/activities/settings/SettingsActivity.java:63]: Please revert changes that only add/remove white space.

File: android/uhabits-android/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -135,8 +136,17 @@
...
+        <item name="highContrastTextColor">@color/grey_200</item>
+        <item name="mediumContrastTextColor">@color/grey_700</item>
         <item name="lowContrastTextColor">@color/grey_900</item>
+
         <item name="highContrastReverseTextColor">@color/black</item>
+        <item name="mediumContrastReverseTextColor">@color/grey_900</item>
+        <item name="lowContrastReverseTextColor">@color/grey_700</item>
+
+        <item name="android:textColor">@color/grey_200</item>
+        <item name="textColorAlertDialogListItem">@color/grey_100</item>
```
Comment (iSoron) [inline android/uhabits-android/src/main/res/values/styles.xml:148]: What problem are you trying to solve with these changes? Please include screenshots.

**Review by iSoron — CHANGES_REQUESTED** [review]
Thank you for the pull request, @Nirvyz. Please see some comments below.

File: android/uhabits-android/src/main/java/org/isoron/uhabits/activities/AndroidThemeSwitcher.kt [hunk 4]
```diff
@@ -19,15 +19,16 @@
...
+import android.app.Activity
+import android.content.Context
+import android.content.res.Configuration.UI_MODE_NIGHT_MASK
+import android.content.res.Configuration.UI_MODE_NIGHT_YES
+import android.os.Build.VERSION.SDK_INT
+import androidx.core.content.ContextCompat
+import org.isoron.androidbase.activities.ActivityScope
+import org.isoron.uhabits.R
+import org.isoron.uhabits.core.preferences.Preferences
+import org.isoron.uhabits.core.ui.ThemeSwitcher
```
[the remark this answers is not in the document]
Comment (mh166) [reply android/uhabits-android/src/main/java/org/isoron/uhabits/activities/AndroidThemeSwitcher.kt:31]: Hi @iSoron, I'm no Java/Kotlin developer myself and have nothing to do with this pull request. But I have a question, just out of curiosity: why is it that you prefer wildcard imports? Is it just a personal preference (which would be fine, mind you) or is there more reason for it? To me (again: no Java dev here) it looks like using wildcard imports would import more than what might be needed, resulting in a larger binary. But I might be terribly wrong here. 😅

On a sidenote: as this project apparently draws the attention of quite a few contributors (which is great! 🥳), maybe it would be a good idea to add a [CONTRIBUTING.md](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/setting-guidelines-for-repository-contributors). This could probably save you some time. 😊

[the remark this answers is not in the document]
Comment (iSoron) [reply android/uhabits-android/src/main/java/org/isoron/uhabits/activities/AndroidThemeSwitcher.kt:31]: @mh166 The main reason for using wildcard imports here is to keep the code consistent with the rest of our code case. Both ways are valid and lead to exactly bytecode. Our production APK also goes through proguard, which removes not only unused classes, but also unused methods and attributes. For new code bases, I prefer wildcard imports because they work just fine and are much less verbose.

Thanks for the CONTRIBUTING.md suggestion. We already have some information in the wiki section, but it could be improved and should probably be moved to a file, which is easier to find.

File: android/uhabits-android/src/main/res/values/styles.xml (hunk 3 above)
Comment (Nirvyz) [reply android/uhabits-android/src/main/res/values/styles.xml:148]: In dark theme settings, for instance when we want to change the first day of week, a dialog opens up. The font color within this dialog is dark and hard to see. We suggest a lighter font color for this part of the app.

File: android/uhabits-android/src/main/java/org/isoron/uhabits/activities/AndroidThemeSwitcher.kt [hunk 5]
```diff
@@ -54,6 +55,14 @@ constructor(
...
     override fun applyLightTheme() {
         activity.setTheme(R.style.AppBaseTheme)
+
+        if(isEdgeToEdgeEnabled(activity.baseContext) == 2) {
+            activity.window.navigationBarColor =
+                    ContextCompat.getColor(activity, R.color.grey_200)
+        } else {
+            activity.window.navigationBarColor =
+                    ContextCompat.getColor(activity, R.color.transparent)
+        }
```
[the remark this answers is not in the document]
Comment (Nirvyz) [reply android/uhabits-android/src/main/java/org/isoron/uhabits/activities/AndroidThemeSwitcher.kt:65]: When using navigation gestures with a navigation bar on Android 10+, the base color of the navbar is inconsistent with the background and makes the experience less immersive for the user. This is why we make it transparent.

File: android/uhabits-android/src/main/java/org/isoron/uhabits/activities/AndroidThemeSwitcher.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Nirvyz) [reply android/uhabits-android/src/main/java/org/isoron/uhabits/activities/AndroidThemeSwitcher.kt:31]: Thank you for the feedbacks. However, we are still new to android programming and we are not sure to understand well what are wildcard imports. I tried to change the import in the latest commits. Please tell me if the imports are correct.

File: android/uhabits-android/src/main/java/org/isoron/uhabits/activities/AndroidThemeSwitcher.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (iSoron) [reply android/uhabits-android/src/main/java/org/isoron/uhabits/activities/AndroidThemeSwitcher.kt:65]: Please see screenshot below. Your changes are causing the navigation buttons in Android 9 to become white, together with the navigation bar.

Screenshot_1607567664

**Comment by iSoron** [comment]
Merged partially in f80fb35e. Thanks once again for the pull request, @Nirvyz!
